dfget/core/downloader/p2p_downloader/piece.go

```diff
@@ -57,20 +58,15 @@ type Piece struct {
 	// length the length of the content.
 	length int64
 
-	// autoReset automatically reset content after reading.
-	autoReset bool
+	// writerNum record the writer number which will write this piece.
+	writerNum int32
 }
 
 // RawContent returns raw contents,
 // If the piece has wrapper, and the piece content will remove the head and tail.
 func (p *Piece) RawContent(noWrapper bool) *bytes.Buffer {
 	contents := p.Content.Bytes()
 	length := len(contents)
-	defer func() {
-		if p.autoReset {
-			p.ResetContent()
```

Comment:
Why not invoke `TryResetContent` here to automatically release buffer?

Reply:
> Why not invoke `TryResetContent` here to automatically release buffer?

we should release the buffer after the io.Copy() finishes, otherwise the same buffer may be reused for writing at the same time.  #1415

Reply:
How to provide a method `WriteTo` in `piece` struct to write the raw content to writer and then release the buffer if necessary, just like :
```
n, e := piece.WriteTo(writer, noWrapper)
```

Reply:
that's a good idea,I will try it.